phpBB/phpbb/notification/type/base.php

```diff
@@ -471,22 +469,47 @@ protected function check_user_notification_options($user_ids = false, $options =
 			{
 				$rowset[$row['user_id']] = array();
 			}
+			$rowset[$row['user_id']][$row['method']] = $row['notify'];
 
-			$rowset[$row['user_id']][] = $row['method'];
+			if (!isset($output[$row['user_id']]))
+			{
+				$output[$row['user_id']] = array();
+			}
+			if ($row['notify'])
+			{
+				$output[$row['user_id']][] = $row['method'];
+			}
 		}
 
 		$this->db->sql_freeresult($result);
 
+		$default_methods = $this->notification_manager->get_default_methods();
```

Comment:
Can you move the assignment to default_methods outside the while loop? This should be called exactly ones and not multiple times inside a loop.